src/solve.jl

```diff
@@ -96,6 +96,16 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,
         error("This solver is not able to use mass matrices.")
     end
 
+    if alg isa OrdinaryDiffEqRosenbrockAdaptiveAlgorithm &&
+            prob.f.mass_matrix isa AbstractMatrix &&
+            all(isequal(0), prob.f.mass_matrix)
+        # technically this should also warn for zero operators but those are hard to check for
+        alg isa Union{Rosenbrock23, Rosenbrock32} && error("Rosenbrock23 and Rosenbrock32 require at least one differential variable to produce valid solutions")
+        if dense && verbose
```

Comment:
Should also warn if `saveat` is not empty.

Reply:
why is `saveat` related?

Reply:
`saveat` uses interpolation

Reply:
oh, so you want `if (dense || !isempty(saveat)) && verbose?

Reply:
yes